coding/javascript.md

```diff
@@ -37,6 +37,26 @@ We use [yarn](https://yarnpkg.com/en) for dependency management.
 
 We use the [jest](https://jestjs.io) unittest framework.
 
+#### General Testing Principals
+  * Avoid setting up state in `beforeEach`. It should be clear from reading a test in isolation what state it is in.
+  * Use test factories to reduce boilerplate for commonly instantiated objects.
+  * Don't test external code/libraries.
+
+#### Testing React
+  * Tests should be collocated with their component in the same directory (MyComponent.js should have a corresponding MyComponent.test.js)
+  * If a component seems difficult to test, it might indicate that it needs to be further decomposed into smaller components.
+  * Unittests should test component behaviour, rather than explicit rendering. Typically you want to test state changes,
+    that handlers are called with appropriate arguments, and any internal logic.
+  * [Jest snapshots](https://jestjs.io/docs/en/snapshot-testing) are ideal for catching regressions in component output/rendering. If you are reviewing a branch, do take care to inspect the snapshot diffs.
+  * Favour testing components in isolation (use enzyme's `shallow` over `mount` where possible).
+
+##### Redux
```

Comment:
I reckon all reducers need a test to ensure they are pure functions as this is where stuff can easily go wrong with Redux, but isn't enforced by the library itself. Might be worth adding a note about that here.